qiita_db/metadata_template/base_metadata_template.py

```diff
@@ -641,44 +601,27 @@ def _add_common_creation_steps_to_queue(cls, md_template, obj_id,
             The queue where the SQL statements will be added
         """
         cls._check_subclass()
+
         # Get some useful information from the metadata template
         sample_ids = md_template.index.tolist()
-        num_samples = len(sample_ids)
-        headers = list(md_template.keys())
-
-        # Get the required columns from the DB
-        db_cols = sorted(get_table_cols(cls._table, conn_handler))
-        # Remove the sample_id and _id_column columns
-        db_cols.remove('sample_id')
-        db_cols.remove(cls._id_column)
+        headers = sorted(md_template.keys().tolist())
```

Comment:
At this point is it assumed that the md_template has been validated and clean (I think it's a safe assumption since this is a private function). Those checks are done in `_clean_validate_template`. That sounds reasonable?